src/parsing/astHelper.ts

```diff
@@ -171,7 +171,7 @@ function followPath(ast: IAST, path: string[]): IAST | null {
 /**
  * Get the value of the given attribute
  */
-function getAttribute(ast: IAST, attributeName: string): string | null {
+function getAttribute(ast: IAST, attributeName: string): string | SequenceType | null {
```

Comment:
you can even make an overload like this:
```
function getAttribute(ast: IAST, attributeName: 'type'): SequenceType | null
function getAttribute(ast: IAST, attributeName: Omit<string, 'type'>): SequenceType | null
function getAttribute(ast: IAST, attributeName: string): string | SequenceType | null {
```

That way you need fewer casts later on.

(not sure about the second line, might be needed, might not be needed)

Reply:
I'm not really sure about the Omit type, never seen it before. What would it do in this case?

Reply:
The omit would cause typescript to never consider the second declaration when called with the string `type`, which makes the return type always be `SequenceType` or null. But I suspect you should not need it